monero/wallet_client.go

```diff
@@ -221,49 +250,211 @@ func (c *walletClient) WaitForReceipt(req *WaitForReceiptRequest) (*wallet.Trans
 }
 
 func (c *walletClient) Transfer(
+	ctx context.Context,
 	to mcrypto.Address,
 	accountIdx uint64,
 	amount *coins.PiconeroAmount,
-) (*wallet.TransferResponse, error) {
+	numConfirmations uint64,
+) (*wallet.Transfer, error) {
 	amt, err := amount.Uint64()
 	if err != nil {
 		return nil, err
 	}
-	return c.wRPC.Transfer(&wallet.TransferRequest{
+	amountStr := amount.AsMoneroString()
+	log.Infof("Transferring %s XMR to %s", amountStr, to)
+	reqResp, err := c.wRPC.Transfer(&wallet.TransferRequest{
 		Destinations: []wallet.Destination{{
 			Amount:  amt,
 			Address: string(to),
 		}},
 		AccountIndex: accountIdx,
 	})
+	if err != nil {
+		log.Warnf("Transfer of %s XMR failed: %s", amountStr, err)
+		return nil, fmt.Errorf("transfer failed: %w", err)
```

Comment:
don't think we need the warn log and returned error?

Reply:
Obviously, it can be taken too far, and I removed a lot of warnings based on your feedback, but if something super bad happens that must be logged, it is good to have some possibly redundant logging closest to the failure point, especially if the early points have additional context they can provide that would be too verbose pass up the error chain.

For errors that relate to things that might cause loss of funds (or losing track of funds that might be recoverable), I think we want as much context as possible and minor redundancy doesn't hurt.